codegen/src/main/scala/overflowdb/schema/Schema.scala

```diff
@@ -139,6 +139,15 @@ class NodeType(name: String, comment: Option[String], schemaInfo: SchemaInfo)
   extends AbstractNodeType(name, comment, schemaInfo) with HasOptionalProtoId  {
   protected val _containedNodes: mutable.Set[ContainedNode] = mutable.Set.empty
 
+  private var _primaryKey: Option[Property[_]] = None
+
+  def primaryKey(p: Property[_]): this.type = {
+    this._primaryKey = Some(p)
```

Comment:
```suggestion
    this._primaryKey = Option(p)
```
Just in case a user has the clever idea to pass `null` here. `Some(null)` is the analogous of shooting into one's foot. `Option(null) == None`